handlers/register.go

```diff
@@ -42,10 +41,16 @@ func Register(mainRouter *mux.Router, cfg config.Config) {
 	getPaymentRouter := rootPaymentRouter.PathPrefix("/{payment_id}").Subrouter()
 	getPaymentRouter.HandleFunc("", HandleGetPaymentSession).Methods("GET").Name("get-payment")
 
-	// All private endpoints need  payment and user auth, so needs to be it's own subrouter
-	privateRouter := mainRouter.PathPrefix("/private").Subrouter()
-	privateRouter.HandleFunc("/payments/{payment_id}", HandlePatchPaymentSession).Methods("PATCH").Name("patch-payment")
-	privateRouter.HandleFunc("/payments/{payment_id}/external-journey", HandleCreateExternalPaymentJourney).Methods("POST").Name("create-external-payment-journey")
+	// payment-details endpoint needs it's own interceptor
+	paymentDetailsRouter := mainRouter.PathPrefix("/private/payments/{payment_id}/payment-details").Subrouter()
+	paymentDetailsRouter.HandleFunc("", HandleGetPaymentDetails).Methods("GET").Name("get-payment-details")
+
+	// All private endpoints need  payment and user auth, and due to router limitations of applying interceptors, need their own subrouters
+	privatePatchRouter := mainRouter.PathPrefix("/private/payments/{payment_id}").Subrouter()
+	privatePatchRouter.HandleFunc("", HandlePatchPaymentSession).Methods("PATCH").Name("patch-payment")
+
+	privateJourneyRouter := mainRouter.PathPrefix("/private/payments/{payment_id}/external-journey").Subrouter()
+	privateJourneyRouter.HandleFunc("/payments/{payment_id}/external-journey", HandleCreateExternalPaymentJourney).Methods("POST").Name("create-external-payment-journey")
```

Comment:
Path here should be empty; the handler function executes for:
`/private/payments/{payment_id}/external-journey/payments/{payment_id}/external-journey`